modules/internal/ExternalString.chpl

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2004-2019 Cray Inc.
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+module ExternalString {
+  use CPtr;
+
+  //
+  // Use this as a convenient way to get a handle to "char*" during resolution.
+  //
+  type chpl__exportTypeCharPtr = c_ptr(c_char);
```

Comment:
Can you add a TODO mentioning that this shouldn't be necessary, and that the compiler should figure this out itself?

Reply:
How do you mean? As in, eventually I should just be able to query for the type `c_ptr(c_char)` as I would other primitive types?

Reply:
Not necessarily as a primitive type, but that the compiler's architecture should be sufficiently flexible/capable such that you can resolve the string "c_ptr(c_char)" within the compiler itself during resolution.